src/Vehicle/Vehicle.cc

```diff
@@ -551,8 +551,11 @@ Vehicle::~Vehicle()
 void Vehicle::prepareDelete()
 {
     if(_cameras) {
-        delete _cameras;
```

Comment:
FYI: For Qt you almost never want to use c++ delete. It is way safer to alway suse ->deleteLater due to various QObject references.

Reply:
I was thinking on this as well but It looked safer to keep the current tested behavior. Otherwise there is need for investigation: will the `control loop` run so that the object is destroyed? is there any side effect for the object living after the `Vehicle` is deleted?

Reply:
I don't quite get what this prepareDelete thing is supposed to do and why it is needed? If the list of cameras is associated with the vehicle then why isn't the QObject for it parented to Vehicle? Then it will be destroyed when the Vehicle goes away and you don't need any of this prepareDelete thing.